python/tach/parsing/modules.py

```diff
@@ -22,12 +25,46 @@ def find_duplicate_modules(modules: list[ModuleConfig]) -> list[str]:
     return duplicate_module_paths
 
 
-def build_module_tree(source_root: Path, modules: list[ModuleConfig]) -> ModuleTree:
+def canonical_form(cycle: list[str]) -> list[str]:
+    """Rotate cycle to start with the smallest element."""
+    min_index = cycle.index(min(cycle))
+    return cycle[min_index:] + cycle[:min_index]
+
+
+def find_cycles(
+    modules: list[ModuleConfig],
+) -> list[list[str]]:
+    graph = nx.DiGraph()  # type: ignore
+    # Add nodes
+    for module in modules:
+        graph.add_node(module.path)  # type: ignore
+
+    # Add dependency edges
+    for module in modules:
+        for dependency in module.depends_on:
+            graph.add_edge(module.path, dependency)  # type: ignore
```

Comment:
small nit - it looks like building the networkx Digraph is something we're doing in multiple places now - it might make sense to have it be a property or function of `ProjectConfig`, so we're not duplicating code to build the graph

Reply:
Makes sense @caelean . Just one doubt here the module list we are using is validated using `fs.validate_project_modules` in `check.py:154`

If we make it a property of ProjectConfig it will use original modules not the validated ones.

```py
@property
def dependency_graph(self) -> nx.DiGraph:
    graph = nx.DiGraph()  # type: ignore
    # Add nodes
    for module in self.modules:
        graph.add_node(module.path)  # type: ignore

    # Add dependency edges
    for module in self.modules:
        for dependency in module.depends_on:
            graph.add_edge(module.path, dependency)  # type: ignore
    
    return graph
```

I created this property!